src/utils/common.js

```diff
@@ -0,0 +1,28 @@
+/*  
+    classNames example usage:
+    <div className={classNames(['base-class', {'second' : someFlag, 'third' : true}])}></div>
+*/
+const classNames = (names) => {
+    let isArray = Array.isArray;
+    let i;
+    let len;
+    let tmp = typeof names;
+    let out = "";
+
+    if (tmp === "string" || tmp === "number") {
+        return names || "";
+    }
+    if (isArray(names) && names.length > 0) {
+        for (i = 0, len = names.length; i < len; i++) {
+            if ((tmp = classNames(names[i])) !== "") out += (out && " ") + tmp;
```

Comment:
I think it's a bit confusing  to build string directly vs using map/filter functions and `.join()`

```
if (isArray(names) && names.length > 0) {
      return names.map(name => classNames(name)).filter(name => !!name).join(" ");
}

return Object.keys().filter(key => names[key]).join(" ");
```

Reply:
💯